pkgs/top-level/splice.nix

```diff
@@ -37,23 +37,21 @@ let
       inherit name;
       value = let
         defaultValue = mash.${name};
-        buildValue = buildPkgs.${name};
-        runValue = runPkgs.${name};
+        # `or {}` is for the non-derivation attsert splicing case, where `{}` is the identity.
```

Comment:
So I was wrong, we did need some `or {}` for a different branch. `*Pkgs` should always evaluate to weak head normal form, so this is sufficient.